pkg/apis/eventing/v1alpha1/condition_status_type.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ConditionType string
+
+const (
+	// ConditionReady specifies that the resource is ready.
+	ConditionReady ConditionType = "Ready"
+)
+
+// ConditionStatus describes the state of a resource at a point in time.
+type ConditionStatus struct {
```

Comment:
@mattmoor and @jonjohnsonjr are looking at implementing a generic "Condition" library in `pkg` -- we should either use that or continue to use per-type statuses, but not some in-between state where we'll still have to move, but have a set of ConditionTypes that don't fit any one object.

Reply:
https://github.com/knative/pkg/pull/66